crates/hir-ty/src/infer/closure.rs

```diff
@@ -253,6 +250,10 @@ impl CapturedItem {
         self.kind
     }
 
+    pub fn spans(&self) -> SmallVec<[MirSpan; 3]> {
+        self.span_stacks.iter().map(|stack| *stack.last().expect("empty span stack")).collect()
```

Comment:
Hm, if only the tails are used couldn't we transform that on construction to save some memory?

Reply:
No, because when a closure contains a closure we extend the capture list of the outer with the capture list of the inner, and then we may truncate them if the outer closure uses a more general borrow (less projections), so we need to be able to truncate them, which means saving all spans.